src/Aspire.Hosting/Publishing/PublishingExtensions.cs

```diff
@@ -8,10 +8,10 @@
 namespace Aspire.Hosting.Publishing;
 
 /// <summary>
-/// Extension methods for <see cref="PublishingStep"/> and <see cref="PublishingTask"/> to provide direct operations.
+/// Extension methods for <see cref="IPublishingStep"/> and <see cref="IPublishingTask"/> to provide direct operations.
 /// </summary>
 [Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
-public static class PublishingExtensions
+internal static class PublishingExtensions
```

Comment:
Keep this class public.

Update it to use the IPublishingStep and IPublishingTask interfaces.